### PR TITLE
step 1: new hackstack modal

### PR DESCRIPTION
**Pull request by smallst**
fix ENG-1222

image

i think this code would be live after Nov.7 anyway so do not add checking now

## Release Notes

- **New Features**
	- Introduced new promotional components for the AI HackStack, enhancing user engagement with updated content and visuals.
	- Added a new `HackstackAutoPromotion` component to the homepage for dynamic promotion display.
	- Added new sections in localization for AI features, HackStack, and curriculum enhancements.
	- Launched a new `HackstackPromotionModal` component for improved promotional displays.

- **Improvements**
	- Enhanced modal closing logic for better user experience.
	- Extensive updates to English localization, improving clarity and engagement in educational content.
	- Updated promotional images in the modal for a refreshed look.
	- Modified the Top Banner to focus on AI HackStack, with updated links and tracking.

- **Bug Fixes**
	- Resolved issues with image sources in the Hackstack promotion modal.

### CODE REVIEW

File: app/views/ai/HackstackAutoPromotion.vue [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<script>
+import ModalDynamicPromotion from 'ozaria/site/components/teacher-dashboard/modals/ModalDynamicContent'
+import HackstackPromotionComponent from './HackstackPromotionComponent'
+export default {
+  components: {
+    ModalDynamicPromotion,
+    HackstackPromotionComponent,
+  },
+  methods: {
+    openModalDynamicPromotion () {
```
Comment (adamkecskes) [inline app/views/ai/HackstackAutoPromotion.vue:10]: what is this method for?

File: app/views/ai/HackstackAutoPromotion.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+<template>
+  <div class="promotion-modal">
+    <ModalDynamicPromotion
+      ref="modal"
+      seen-promotions-property="new-hackstack-promotion-modal"
+    >
+      <template #content>
+        <HackstackPromotionComponent />
+      </template>
```
Comment (adamkecskes) [inline app/views/ai/HackstackAutoPromotion.vue:25]: we should handle the close coming from the "learn more" button here
```suggestion
        <HackstackPromotionComponent
          @close="onClose"
        />
```

File: app/views/ai/HackstackAutoPromotion.vue [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import HackstackPromotionComponent from './HackstackPromotionComponent'
+export default {
+  components: {
+    ModalDynamicPromotion,
+    HackstackPromotionComponent,
+  },
+  methods: {
+    openModalDynamicPromotion () {
+      this.modalDynamicPromotion = true
+    },
```
Comment (adamkecskes) [inline app/views/ai/HackstackAutoPromotion.vue:12]: and we could call the modal's onClose like this:
```suggestion
    onClose () {
      this.$refs.modal.onClose()
    },
```

File: app/views/ai/HackstackPromotionComponent.vue [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  data () {
+    return {
+      videoId: '827b895ec6a340f0a701c456649e274a',
+    }
+  },
+  methods: {
+    onLearnMore () {
+      this.$refs.modal.onClose()
+      this.trackEvent('Hackstack Promo Modal: Learn more clicked', { category: 'Hackstack' })
+    },
```
[the remark this answers is not in the document]
Comment (adamkecskes) [reply app/views/ai/HackstackPromotionComponent.vue:63]: it seems codeRabbit is right here. we should emit the `close` here.

**Review by mrfinch — COMMENTED** [review]
reduce the size of video for modal to be able to fit the screen
Screenshot 2024-11-06 at 4 39 41 PM

File: app/views/ai/HackstackAutoPromotion.vue (hunk 1 above)
Comment (smallst) [reply app/views/ai/HackstackAutoPromotion.vue:10]: thank you for find out some unused function

**Review by mrfinch — APPROVED** [review]

**Pull request merged**